### PR TITLE
[CVE] Handle invalid query, index and date in vega charts filter handlers

### PR DESCRIPTION
**Pull request by bandinib-amzn**
### Description

Potential way to prevent XSS vulnerability discovered in the Vega charts OSD integration.

CVE link:
https://nvd.nist.gov/vuln/detail/CVE-2022-23713

Signed-off-by: Bandini Bhopi <bandinib@amazon.com>

Testing:

Sanity testing by creating Vega visualization from [Vega Example Gallery](https://vega.github.io/vega/examples/)
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
    - [X] `yarn test:jest`
    - [X] `yarn test:jest_integration`
    - [X] `yarn test:ftr`
- [ ] New functionality has been documented.
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: packages/osd-std/src/validate_object.test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+/*
```
Comment (kavilla) [inline packages/osd-std/src/validate_object.test.ts:1]: If this is a new file, then the new license header is all that is needed like this: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/dev-tools/get-version.sh#L3

File: packages/osd-std/src/validate_object.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
```
Comment (kavilla) [inline packages/osd-std/src/validate_object.ts:13]: same as above

File: src/plugins/vis_type_vega/public/data_model/utils.test.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
```
Comment (kavilla) [inline src/plugins/vis_type_vega/public/data_model/utils.test.js:13]: same as above

**Review by kavilla — COMMENTED** [review]
Looks pretty good!

File: packages/osd-std/src/validate_object.test.ts (hunk 1 above)
Comment (bandinib-amzn) [reply packages/osd-std/src/validate_object.test.ts:1]: I updated license header.

File: packages/osd-std/src/validate_object.ts (hunk 2 above)
Comment (bandinib-amzn) [reply packages/osd-std/src/validate_object.ts]: updated

File: src/plugins/vis_type_vega/public/data_model/utils.test.js (hunk 3 above)
Comment (bandinib-amzn) [reply src/plugins/vis_type_vega/public/data_model/utils.test.js]: Updated.

**Review by manasvinibs — APPROVED** [review]
LGTM

**Review by ZilongX — APPROVED** [review]

**Pull request merged**